server/routes/bulkOffchain.ts

```diff
@@ -15,266 +15,293 @@ import { RoleInstance } from '../models/role';
+          //     : [];
+          //   const chain_entities = JSON.parse(t.chain_entities[0]).id
+          //     ? t.chain_entities.map((c) => JSON.parse(c))
+          //     : [];
+          //   const linked_threads = JSON.parse(t.linked_threads[0]).id
+          //     ? t.linked_threads.map((c) => JSON.parse(c))
+          //     : [];
+          //   const last_edited = getLastEdited(t);
 
-        const monthlyComments = await models.OffchainComment.findAll({ where, include: [ models.Address ] });
-        const monthlyThreads = await models.OffchainThread.findAll({
-          where,
-          attributes: { exclude: ['version_history'] },
-          include: [{ model: models.Address, as: 'Address' }],
-        });
+          //   const data = {
+          //     id: t.thread_id,
+          //     title: t.thread_title,
+          //     url: t.url,
+          //     body: t.body,
+          //     last_edited,
+          //     kind: t.kind,
+          //     stage: t.stage,
+          //     read_only: t.read_only,
+          //     pinned: t.pinned,
+          //     community: t.thread_community,
+          //     chain: t.thread_chain,
+          //     created_at: t.thread_created,
+          //     collaborators,
+          //     linked_threads,
+          //     chain_entities,
+          //     snapshot_proposal: t.snapshot_proposal,
+          //     offchain_voting_options: t.offchain_voting_options,
+          //     offchain_voting_votes: t.offchain_voting_votes,
+          //     offchain_voting_ends_at: t.offchain_voting_ends_at,
+          //     Address: {
+          //       id: t.addr_id,
+          //       address: t.addr_address,
+          //       chain: t.addr_chain,
+          //     },
+          //   };
+          //   if (t.topic_id) {
+          //     data['topic'] = {
+          //       id: t.topic_id,
+          //       name: t.topic_name,
+          //       description: t.topic_description,
+          //       communityId: t.topic_community,
+          //       chainId: t.topic_chain,
+          //       telegram: t.telegram,
+          //     };
+          //   }
+          //   return data;
+          // });
+        } catch (e) {
+          console.log(e);
+          reject(new Error('Could not fetch threads, comments, or reactions'));
+        }
+      }),
+      // admins
+      models.Role.findAll({
+        where: chain
+          ? {
+              chain_id: chain.id,
+              permission: { [Op.in]: ['admin', 'moderator'] },
+            }
+          : {
+              offchain_community_id: community.id,
+              permission: { [Op.in]: ['admin', 'moderator'] },
+            },
+        include: [models.Address],
+        order: [['created_at', 'DESC']],
+      }),
+      // most active users
+      new Promise(async (resolve, reject) => {
+        try {
+          const thirtyDaysAgo = new Date(
+            (new Date() as any) - 1000 * 24 * 60 * 60 * 30
```

Comment:
I'm confused. What's the issue with this solution, and why does using Date.now beat new Date? My understanding has been that they're functionally equivalent, since both can be manipulated as milliseconds.

Heads up I didn't write this code, I just linted it. This has been our